### DISCUSSION TITLE
Which 3rd party integration would you like to see/have?

### DISCUSSION THREAD

**Discussion by pglombardo** [pglombardo/PasswordPusher#403] [Ideas]
Password Pusher can be integrated an a large variety of places but we need to know which places are most requested.

Please **upvote integrations** or **add your integration idea below**!

**Comment by pglombardo** [comment]
Official Slack Integration
https://slack.com/integrations

**Comment by pglombardo** [comment]
Microsoft Teams Application
https://docs.microsoft.com/en-us/microsoftteams/platform/overview

**Reply by pglombardo** [reply]
Teams is hard 😢   And after the Microsoft layoffs, it seems large parts of the SDK are unmaintained.  Going to revisit over the holidays.

**Comment by pglombardo** [comment]
Zendesk Application
https://developer.zendesk.com/documentation/apps/build-an-app/building-a-server-side-app/part-1-core-concepts/

**Comment by pglombardo** [comment]
Halo Service Desk Integration
https://haloservicedesk.com/halo-service-desk-integrations-2021/

**Reply by PC-Bjorn** [reply]
Any news on this? :)

**Reply by pglombardo** [reply]
Hi @PC-Bjorn et. al. - I've looked at Halo multiple times and they don't seem to have an open integration system.  It seems I'll have to get in touch with the company to see if they'll accept a Password Pusher integration.  I've started a trial with them and will send an inquiry soon.  If you or anyone has any ideas on this I'd love to hear them.

**Reply by martyn-gg** [reply]
Being able to integrate with our self-hosted docker instance would be fantastic!

**Reply by LiamMcCreadyONeil** [reply]
Here to add a vote for this. 
I have an on-premise Halo sandbox if you are needing somewhere to test long term.

**Reply by pglombardo** [reply]
I got in touch with the company but couldn't get past the sales people.  They promised to get back to me but never did.  Then it was months of robot emails asking if I want to talk/purchase a license.  😄 

When I get another chance, maybe I'll reach out to the developers on LinkedIn as a different strategy.

> I have an on-premise Halo sandbox if you are needing somewhere to test long term.

This is much appreciated but I would first need a copy of their software and the install/API documentation to see how/where to best integrate.  After that though, having a real user periodically test would be a big help.

**Reply by qcomer** [reply]
@pglombardo  We are Halo onboarding partners and consultants. Would be happy to get you sandbox access :) We use PWPush within Halo runbooks and stuff. You'd have full swagger and API access also.

**Reply by pglombardo** [reply]
Hi @qcomer - this would be a huge help!  Could you email me at pglombardo@pwpush.com?  We can then kick off the process to get something built.

**Reply by MrNickIE** [reply]
Any news on this?  Desperate to get this integrated.  Have paid PWPush sub.

**Reply by robertchilds87** [reply]
@pglombardo Did you manage to get anywhere with HaloPSA and PWpush after discussions with @qcomer ?

**Comment by Cristianmutis** [comment]
Hola, me gustaria saber de que manera se puede integrar con Slack, debido a que he intentado con el slack command y no funciona en mi instancia privada. saludos.

**Reply by pglombardo** [reply]
La integración de Slack no ha sido tocada o actualizada en algún tiempo.  Veré si puedo probarla y actualizarla más temprano que tarde.

**Comment by xpufx** [comment]
I am kind of surprised nobody said it yet but OwnCloud/NextCloud integration sounds like a very natural fit. I am imagining sharing files with 3rd parties using pwpush. It would also work really well is an embedded app there but I think those need to be written in PHP.

**Comment by grulicht** [comment]
## 📢 New Terraform Provider for Password Pusher (`pwpush`)

Hi Password Pusher team 👋,

I'm excited to share that I've created an **official Terraform provider** for Password Pusher!

🔗 **GitHub:** [grulicht/terraform-provider-pwpush](https://github.com/grulicht/terraform-provider-pwpush)  
🌍 **Terraform Registry:** [registry.terraform.io/providers/grulicht/pwpush](https://registry.terraform.io/providers/grulicht/pwpush/latest)  
🧱 **OpenTofu:** [search.opentofu.org/provider/grulicht/pwpush](https://search.opentofu.org/provider/grulicht/pwpush/latest)
**Discussions:** https://github.com/pglombardo/PasswordPusher/discussions/3373

Best regards,  
**Tomáš Grulich**  
[https://github.com/grulicht](https://github.com/grulicht)
